salt/channel/client.py

```diff
@@ -594,14 +594,22 @@ def _verify_master_signature(self, payload):
     def _decode_payload(self, payload):
         # we need to decrypt it
         log.trace("Decoding payload: %s", payload)
+        reauth = False
         if payload["enc"] == "aes":
             self._verify_master_signature(payload)
             try:
                 payload["load"] = self.auth.crypticle.loads(payload["load"])
             except salt.crypt.AuthenticationError:
-                yield self.auth.authenticate()
-                payload["load"] = self.auth.crypticle.loads(payload["load"])
-
+                reauth = True
+            if reauth:
+                try:
+                    yield self.auth.authenticate()
+                    payload["load"] = self.auth.crypticle.loads(payload["load"])
+                except salt.crypt.AuthenticationError:
+                    log.error(
+                        "Payload decryption failed even after re-authenticating with master %s",
+                        self.opts["master_ip"],
```

Comment:
```suggestion
                        self.opts["master_ip"],
```
You are assuming that `master_ip` will be set?
[Where does it get set?](https://github.com/search?q=repo%3Asaltstack%2Fsalt+master_ip+path%3A%2F%5Esalt%5C%2F%2F&type=code)

By the user?
Should it instead be `opts["id"]`?

Reply:
Still a valid question

Reply:
The intent here is to let the user know which master the minion is trying to connect to; so we want `opts['master']` or `opts['master_ip']`.
`master_ip` is set in `salt/minion.py::Minion.eval_master` by `salt/minion.py::resolve_dns` and is always set by the time a channel gets created.

Reply:
If that's the case, I'm good.